### ISSUE TITLE
Documentation on shared/static variants confusing

### ISSUE THREAD

**Issue by DavidPoliakoff** [spack/spack#10089]
Hey folks,

I'm looking to update the Caliper package, specifically to add a `static` or `shared` variant. I've seen this is a mildly discussion-provoking topic in issues like #5269 , and that the documentation currently says that if you do this with variants, [to name them `static` or `shared`](https://spack.readthedocs.io/en/latest/packaging_guide.html#variant-names) and default your variant to true.  

Is there a canonical way to do this? Do most packages prefer to name it `shared`, or do people lean towards `static`? What would be best style/practice, or is this one of those places where each package picks its own logic?

**Comment by healther** [comment]
```
$ grep -r shared var/spack/repos/builtin/packages/*/package.py | grep variant | wc -l
77 
$ grep -r static var/spack/repos/builtin/packages/*/package.py | grep variant | wc -l
4
```
I think it's fair to say that the consensus tends to `shared`.

**Comment by citibeth** [comment]
I agree, this is a corner that has not been sufficiently cleaned up.  We've talked a lot about it, but nobody has done the work.

By default, packages should build shared.
